docs/creating_platform_code_review.md

```diff
@@ -15,63 +16,76 @@ A checklist of things to do when you're adding a new platform.
+ 2. Requirement version should be pinned: `"requirements": ['phue==0.8.1']`
+ 3. We no longer want requirements hosted on GitHub. Please upload to PyPi.
+ 4. Requirements should only be imported inside functions. This is necessary because requirements are installed on the fly.
 
 ### 2. Dependencies
 
- 1. If you depend on a component for the connection, add it to your dependencies: `DEPENDENCIES = ['nest']`
+ 1. If you depend on a component for the connection, add it to your dependencies in [`manifest.json`](creating_integration_manifest.md): `"dependencies": ['nest']`. The `DEPENDENCIES` constant is deprecated.
 
 ### 3. Configuration
 
  1. Voluptuous schema present for [configuration validation](development_validation.md)
- 2. Voluptuous schema extends schema from component<br>(e.g., `light.hue.PLATFORM_SCHEMA` extends `light.PLATFORM_SCHEMA`)
+ 2. Voluptuous schema extends schema from component<br>(e.g., `hue.light.PLATFORM_SCHEMA` extends `light.PLATFORM_SCHEMA`)
  3. Default parameters specified in voluptuous schema, not in `setup_platform(...)`
  4. Your `PLATFORM_SCHEMA` should use as many generic config keys as possible from `homeassistant.const`
-    ```python
-    import voluptuous as vol
+ 5. Never depend on users adding things to `customize` to configure behavior inside your platform.
 
-    from homeassistant.const import CONF_FILENAME, CONF_HOST
-    from homeassistant.components.light import PLATFORM_SCHEMA
-    import homeassistant.helpers.config_validation as cv
+```python
+import voluptuous as vol
 
-    CONF_ALLOW_UNREACHABLE = 'allow_unreachable'
-    DEFAULT_UNREACHABLE = False
+from homeassistant.const import CONF_FILENAME, CONF_HOST
+from homeassistant.components.light import PLATFORM_SCHEMA
+import homeassistant.helpers.config_validation as cv
 
-    PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
-        vol.Required(CONF_HOST): cv.string,
-        vol.Optional(CONF_ALLOW_UNREACHABLE,
-                    default=DEFAULT_UNREACHABLE): cv.boolean,
-        vol.Optional(CONF_FILENAME): cv.string,
-    })
-    ```
- 5. Never depend on users adding things to `customize` to configure behavior inside your platform.
+CONF_ALLOW_UNREACHABLE = 'allow_unreachable'
+DEFAULT_UNREACHABLE = False
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_HOST): cv.string,
+    vol.Optional(CONF_ALLOW_UNREACHABLE,
+                default=DEFAULT_UNREACHABLE): cv.boolean,
+    vol.Optional(CONF_FILENAME): cv.string,
+})
+```
 
 ### 4. Setup Platform
 
- 1. Test if passed in info (user/pass/host etc.) works.
+ 1. Verify that the passed in configuration (user/pass/host etc.) works.
  2. Group your calls to `add_devices` if possible.
- 3. If platform adds extra services, format should be `<component>.<platform>_<service name>`.
+ 3. If platform adds extra services, format should be `<domain>.<service name>`. Make sure that your services [verify permissions](auth_permissions.md#checking-permissions).
```

Comment:
```suggestion
 3. If the platform adds extra services, the format should be `<domain>.<service name>`. Make sure that your services [verify permissions](auth_permissions.md#checking-permissions).
```

Reply:
It isn't clear to me what domain we reference here.